### PR TITLE
feat(ops): add unit test infra and a demonstration test to sdk

### PR DESCRIPTION
**Pull request by Orta21**
Ref. https://github.com/metriport/metriport/issues/537

Ref: 537

### Description

Add unit test infra and a demonstration test

### Release Plan

Nothing special

- [ ] Once approved

### CODE REVIEW

File: packages/api-sdk/package.json [hunk 1]
```diff
@@ -44,7 +44,8 @@
     "lint": "npx eslint . --ext .ts",
     "lint-fix": "npm run lint --fix",
     "prettier-fix": "npx prettier '**/*.ts' --write",
-    "test": "echo \"No test specified yet\""
+    "test": "jest --runInBand --detectOpenHandles --passWithNoTests",
+    "test:e2e": "E2E=true jest --runInBand --detectOpenHandles"
```
Comment (leite08) [inline packages/api-sdk/package.json:48]: I don't think we'll need/implement E2E tests on the SDKs, or am I missing something?

File: packages/api-sdk/src/__tests__/shared.ts
Comment (leite08) [inline packages/api-sdk/src/__tests__/shared.ts:1]: Not being used?

File: packages/api-sdk/src/devices/client/__tests__/metriport.test.ts
Comment (leite08) [inline packages/api-sdk/src/devices/client/__tests__/metriport.test.ts:1]: 🤩

File: packages/api-sdk/package.json (hunk 1 above)
Comment (Orta21) [reply packages/api-sdk/package.json:48]: Good point, i copied this from api

File: examples/react-native/App.tsx [hunk 2]
```diff
@@ -6,13 +6,15 @@ import { MetriportWidget } from "@metriport/react-native-sdk";
 export default function App() {
   return (
     <MetriportWidget
-      token="CONNECT_TOKEN"
-      clientApiKey="CLIENT_API_KEY"
+      token="u_TuOJkN_s78vdyWmI8xC"
+      clientApiKey="ck9kMkJzWU5qSS1nQ0wtVzVTenFTOjIxZmE0MzJlLTcyM2ItNGExZC1hM2IyLWJkOWNkNzVhMDcxNw"
```
Comment (leite08) [inline examples/react-native/App.tsx:10]: Are these supposed to be sent to GH?

File: examples/react-native/App.tsx [hunk 3]
```diff
@@ -6,13 +6,15 @@ import { MetriportWidget } from "@metriport/react-native-sdk";
...
       colorMode="dark"
       customColor="green"
-      sandbox={true}
-      providers={["fitbit", "cronometer"]}
+      sandbox={false}
+      // providers={["fitbit", "cronometer"]}
```
Comment (leite08) [inline examples/react-native/App.tsx:14]: left-over?

File: packages/api-sdk/src/devices/client/__tests__/metriport.test.ts [hunk 4]
```diff
@@ -7,7 +6,7 @@ jest.mock("axios");
 describe("getMetriportUserId", () => {
   const userId = "1";
   const apiKey = "FAKE_KEY";
-  const mockedAxios = mocked(axios, true);
+  const mockedAxios = axios as jest.Mocked<typeof axios>;
```
Comment (leite08) [inline packages/api-sdk/src/devices/client/__tests__/metriport.test.ts:9]: Why not `mocked(axios...)`?

`as` only makes it to be treated as such, but it doesn't change its underlying implementation/code

Comment (Orta21) [reply packages/api-sdk/src/devices/client/__tests__/metriport.test.ts:9]: I was getting a typescript error in the build process that it didnt like unless i did something like this

Comment (Orta21) [reply packages/api-sdk/src/devices/client/__tests__/metriport.test.ts:9]: I was saying how the second param in the mocked function should have been an object even though its suppose to be a boolean

Comment (Orta21) [reply packages/api-sdk/src/devices/client/__tests__/metriport.test.ts:9]: @leite08

Comment (Orta21) [reply packages/api-sdk/src/devices/client/__tests__/metriport.test.ts]: Yea i removed boolean and it works now

File: packages/api-sdk/src/devices/client/__tests__/metriport.test.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { mocked } from "jest-mock";
+import axios from "axios";
+import { MetriportDevicesApi } from "../metriport";
+
+jest.mock("axios");
+
+describe("getMetriportUserId", () => {
+  const userId = "1";
+  const apiKey = "FAKE_KEY";
```
Comment (leite08) [inline packages/api-sdk/src/devices/client/__tests__/metriport.test.ts:9]: NABD, but its good to make these dynamic with `faker-js` or `uuidv4` or something.

File: packages/api-sdk/src/devices/client/__tests__/metriport.test.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    jest.spyOn(metriportClient, "getMetriportUserId");
+
+    mockedAxios.post.mockResolvedValueOnce({ data: { userId } });
+
+    const result = await metriportClient.getMetriportUserId(userId);
+
+    expect(mockedAxios.post).toHaveBeenCalledTimes(1);
+    expect(axios.post).toHaveBeenCalledWith(`/user`, null, { params: { appUserId: userId } });
+    expect(metriportClient.getMetriportUserId).toHaveBeenCalledTimes(1);
```
Comment (leite08) [inline packages/api-sdk/src/devices/client/__tests__/metriport.test.ts:25]: I don' think this one is needed, right? Its testing the test.

**Review by leite08 — APPROVED** [review]
LGTM, just a couple nits

**Pull request merged**